Modified transaction edit button on mobile to display "Add New Split"

tempiz opened the pull request:
On the desktop client, a user can split a transaction an endless amount of times. On mobile, it is hard coded to a single split. This change creates an add split button below split transaction boxes on mobile to allow for additional splits.

**Edit:** I was informed infact it is possible to add endless amounts of splits on mobile by using the "Remaining" button at the bottom. This may not be intuitive for new users like myself. This PR adds an "Add Split" button below the split transaction elements on the page for a clear intuitive way to add more splits.

I am open to feedback on this, or closing if other reviewers feel the current implementation is sufficient.

tempiz committed:
Created Add Split Button

youngcw commented:
There is already a button to add a split.  The the one that says how much is left

tempiz committed:
Create 4536.md

tempiz commented:
> There is already a button to add a split. The the one that says how much is left

I may not have explained this right, or im just blind. The goal here is to allow the user to split more than twice. So splitting a transaction three or four ways instead of just splitting twice. On desktop there is an add split button that allows splitting between more than two categories on one transaction. Unless I am blind on mobile, which is possibe lol.

youngcw commented:
You can make many splits on mobile.  The button at the bottom adds more split lines

tempiz commented:
> You can add more than two ways. The button at the bottom adds more split lines

Ahhh I see now. I was clicking it while I didn't assign both other boxes. Sorry for wasting your time!

youngcw commented:
Feel free to try to make that process easier/more intuitive

tempiz commented:
> Feel free to try to make that process easier/more intuitive

My PR here simply adds an additional "Add Split" button below the latest split element. I could see it being advantageous to leave the functionality of the "Remaining" button's ability to split, as well as the addition of this new button. It would still allow the work flow of users who are accustomed to the current method, while new users like myself may find it more obvious with a separate button.

Open to your thoughts.

tempiz committed:
Fix Lint Check Error

tempiz committed:
Merge branch 'master' into mobile-addsplit

tempiz committed:
Fix Release Note Category

tempiz committed:
Merge branch 'mobile-addsplit' of https://github.com/tempiz/actual into mobile-addsplit

tempiz committed:
Merge branch 'master' into mobile-addsplit

joel-jeremy commented:
Maybe we should just update the text on the primary button instead of adding a new one?

- If there is a split with an empty amount, the button should say something like `Enter split amount`
  - This will scroll you to that split form with an empty amount
- If there is no split with an empty amount, the button should say something like `Add new split`
  - This will create an empty split 

It would be great if there's a new button that has something like `Add new split with remaining amount` though.

tempiz commented:
> Quote reply

Not a bad compromise. I should have some time this weekend to look into this.

tempiz committed:
Merge branch 'master' into mobile-addsplit

tempiz committed:
Added add split text to amount left/save button

Removed my add split button and instead added functionality for the button save/amount left button to display "Add New Split" when all amounts are entered.

tempiz commented:
Adjusted PR to match the solution above.

When all splits have an amount, the button says "Add New Split"
If there are empty splits, it says "Amount left:" and scrolls to the empty split.

tempiz committed:
Fix Lint

tempiz committed:
Merge branch 'master' into mobile-addsplit

tempiz committed:
Update release notes

tempiz committed:
Merge branch 'mobile-addsplit' of https://github.com/tempiz/actual into mobile-addsplit

youngcw commented:
This is good. Can the Add Split text also include how much is left?

tempiz committed:
Show remaining on add split button

tempiz commented:
> This is good. Can the Add Split text also include how much is left?

Added the amount left to the add split text.
![image](https://github.com/user-attachments/assets/03dc3c02-02e4-445b-bbec-c3787187e3b7)

tempiz committed:
Fix lint

tempiz committed:
Merge branch 'master' into mobile-addsplit

joel-jeremy commented on packages/desktop-client/src/components/mobile/transactions/TransactionEdit.jsx:
```diff
@@ -209,18 +209,34 @@ function Footer({
           isDisabled={editingField}
           onPress={onClickRemainingSplit}
         >
+          {}
```
Is this empty block needed?
```suggestion
```

joel-jeremy commented on packages/desktop-client/src/components/mobile/transactions/TransactionEdit.jsx:
```diff
@@ -209,18 +209,34 @@ function Footer({
-                : -transaction.error.difference,
+            {!emptySplitTransaction ? (
+              <>
+                <Trans>Add New Split</Trans>
```
This should be sentence case and i think this should use a formatted translation string e.g. `t('Add new split - {{amount}} left', { amount: integerToCurrency(...) })`

joel-jeremy commented on packages/desktop-client/src/components/mobile/transactions/TransactionEdit.jsx:
```diff
@@ -209,18 +209,34 @@ function Footer({
+              </>
+            ) : (
+              <>
+                <Trans>Amount left</Trans>:{' '}
```
Same comment on translation string above

tempiz replied:
Please see latest changes. I was unable to get a formatted translation string above to work, as I kept getting t undefined, but I did clear up the <Trans> to instead set the amount inline in a similar style as above. Let me know if that works.

Thanks!

tempiz committed:
Added requested changes

tempiz committed:
Merge branch 'mobile-addsplit' of https://github.com/tempiz/actual into mobile-addsplit

tempiz committed:
Merge branch 'master' into mobile-addsplit

tempiz committed:
Fix lint

tempiz committed:
Merge branch 'mobile-addsplit' of https://github.com/tempiz/actual into mobile-addsplit

tempiz committed:
Merge branch 'master' into mobile-addsplit

tempiz committed:
Merge branch 'master' into mobile-addsplit
